### PR TITLE
feat: Injection Protection plugin for 3.9

### PR DESCRIPTION
**Pull request by cloudjumpercat**
### Description

 Adding the new Injection Protection plugin docs.

DOCU-4148
### Testing instructions

Preview link: https://deploy-preview-8177--kongdocs.netlify.app/hub/kong-inc/injection-protection/unreleased/

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

### CODE REVIEW

**Comment by cloudjumpercat** [comment]
Hi @brentos !
I started drafting some docs for the Injection Protection plugin, but had a few questions:
* To what degree is the plugin compatible with DB less? Completely, not at all, or partially?
* How do I collect the logs? Where can I view the logs? Are there multiple log types (in the eng brief, I see the log serializer, analytics proto, and Kong error logs)?

File: app/_hub/kong-inc/injection-protection/_metadata/_index.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+name: Injection Protection
+
+publisher: Kong Inc.
+
+type: plugin
+
+categories: 
+  - security
+
+desc: # (Required) Short, one sentence description; max 80 characters.
```
Comment (brentos) [inline app/_hub/kong-inc/injection-protection/_metadata/_index.yml:14]:
```suggestion
desc: Detect and block injection attacks using regular expressions
```

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+---
+nav_title: Overview
+title: Overview
+---
+
+You can use the Injection Protection plugin to detect well known patterns that might be the result of a cross site scripting or other injection type attacks.
```
Comment (brentos) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:6]:
```suggestion
You can use the Injection Protection plugin to detect and block known injection patterns consistent with SQL injection, server-side include injection and more.
```

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+The Injection Protection plugin makes it easier to protect your APIs from cross site scripting or other injection attacks by providing out-of-the box regex matching for common injection attacks. In addition, you can configure custom regex matching as well.
```
Comment (brentos) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:8]:
```suggestion
The Injection Protection plugin makes it easier to protect your APIs from SQL injection or other injection attacks by providing out-of-the box regex matching for common injection attacks. In addition, you can configure custom regex matching as well. 
```

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+title: Overview
+---
+
+You can use the Injection Protection plugin to detect well known patterns that might be the result of a cross site scripting or other injection type attacks.
+
+The Injection Protection plugin makes it easier to protect your APIs from cross site scripting or other injection attacks by providing out-of-the box regex matching for common injection attacks. In addition, you can configure custom regex matching as well. 
+
+The Injection Protection plugin does the following: 
+* Extracts information from request headers, path/query parameters, or the payload body and evaluate that content against pre-defined regular expression
+* Rejects the requests that match the regular expressions with a 400 bad request
```
Comment (brentos) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:12]:
```suggestion
* Rejects the requests that match the regular expressions with a configurable HTTP status code and error message
```

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+* Captures metrics about rejected requests for analytics and reporting
```
Comment (brentos) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:13]:
```suggestion
* Logs information about rejected requests for analytics and reporting
```

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+## How does the Injection Protection plugin work?
+
+Depending on what you have configured in the plugin's config, the Injection Protection plugin does the following:
+
+1. The plugin extracts the specified content (headers, path/query parameters, payload body, etc.) from a client request.
```
Comment (brentos) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:19]:
```suggestion
1. The plugin extracts the specified content (headers, path/query parameters, payload body) from a client request.
```

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+1. The plugin checks the extracted content for matches against the specified pre-defined or custom regex expressions. The regex expressions define patterns that would match well-known cross site scripting or other injection attacks.
```
Comment (brentos) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:20]:
```suggestion
1. The plugin checks the extracted content for matches against the specified pre-defined or custom regex expressions. The regex expressions define patterns that would match well-known injection attacks.
```

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+1. Depending on if the content matches, the plugin does one of the following:
+    * **Regex doesn't match:** The plugin allows the request and sends a `200` status code to the client.
+    * **Regex match:** The plugin prevents the request by sending a `400` status code to the client and sends {{site.base_gateway}} an error log that contains the name of the injection type, what content matched the pattern, and what regex matched the content. You can also configure the plugin to only log matches and allow requests that match to still by proxied.
```
Comment (brentos) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:23]:
```suggestion
    * **Regex match:** The plugin blocks the request by sending a `400` status code to the client and sends {{site.base_gateway}} an error log that contains the name of the injection type, what content matched the pattern, and what regex matched the content. You can also configure the plugin to only log matches and allow requests that match to still by proxied.
```

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    else Regex match
+        Injection Protection plugin->>Consumer: 400 Bad Request
+        Injection Protection plugin->>{{site.base_gateway}}: Logs injection 
+    end
+{% endmermaid %}
+<!--vale on-->
+
+## What pre-defined regex patterns does the plugin provide for injection attacks?
+
+The Injection Protection plugin comes with several pre-built regex patterns that match common injection attacks. You can specify these patterns when you configure the plugin to easily block common attacks.
```
Comment (brentos) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:46]:
```suggestion
The Injection Protection plugin comes with several pre-built regex patterns that match common injection attacks. You can enable or disable these patterns when you configure the plugin to easily block common attacks.
```

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+| Server-side include | `<!--#(include|exec|echo|config|printenv)\s+.*` | Detects scripts injected in HTML pages. |
+| XPath abbreviated | `(/(@?[\w_?\w:\*]+(\[[^]]+\])*)?)+` | Detects intentionally malformed information that is sent to a website with the intention of constructing an XPath query for XML data, specifically in abbreviated syntax. |
+| XPath extended | `/?(ancestor(-or-self)?|descendant(-or-self)?|following(-sibling))` | Detects intentionally malformed information that is sent to a website with the intention of constructing an XPath query for XML data, specifically in the full syntax. |
+| JavaScript | `<\s*script\b[^>]*>[^<]+<\s*/\s*script\s*>` | Detects abritrarily injected JavaScript that is part of a cross site scripting attack and will execute in the browser. 
+| Java exception | `.*?Exception in thread.*` | Detects denial-of-service (DoS) attacks that match Java exception messages. |
+
+## How do I create a custom regex for matching?
+
+You can specify a custom regex for matching by using the `custom_regex_record` parameter in the Injection Protection plugin config. To create a custom regex, you must define the following:
```
Comment (brentos) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:59]:
```suggestion
You can specify a custom regex for matching by using the `custom_injections` parameter in the Injection Protection plugin config. To create a custom regex, you must define the following:
```

**Review by brentos — DISMISSED** [review]

**Comment by lena-larionova** [comment]
>To what degree is the plugin compatible with DB less? Completely, not at all, or partially?

Assuming full compatibility unless proven otherwise. I currently can't see any reason that this plugin would have issues in DB-less or hybrid mode.

> How do I collect the logs? Where can I view the logs? Are there multiple log types (in the eng brief, I see the log serializer, analytics proto, and Kong error logs)?

I think your list in the intro is currently sufficient. I wouldn't mind seeing a sample log entry, but it shouldn't be an issue if we can't get that for release time.

File: app/_hub/kong-inc/injection-protection/how-to/_index.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+        --header "accept: application/json" \
+        --header "Content-Type: application/json" \
+        --data '
+        {
+            "name": "injection-protection",
+            "config": {
+                "custom_injections": [
+                    {
+                        "name": "boolean_based_sql_injection",
+                        "regex": "(\\b(OR|AND)\\s+\\d{1,3}=\\d{1,3}\\s*(--|#|\\b))|(\\b(OR|AND)\\s+1=1\\s*(--|#|\\b))|(\\b(OR|AND)\\s+1=0\\s*(--|#|\\b))"
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/injection-protection/how-to/_index.md:40]: ChatGPT came up with this regex. Is the regex correct for this use case?

File: app/_hub/kong-inc/injection-protection/how-to/_index.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    ```
+
+2. Create a route:
+    ```sh
+    curl -i -X POST http://localhost:8001/services/ip_example_service/routes \
+        --data 'paths[]=/ip_mock' \
+        --data name=ip_route
+    ```
+
+3. Enable the Injection Protection plugin on the route:
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/injection-protection/how-to/_index.md:28]: ChatGPT helped me with the config, but I tested it and it was enabled on the route successfully.

File: app/_hub/kong-inc/injection-protection/how-to/_index.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    ```
+
+## Validate
+
+Verify that the Injection Protection plugin detects the boolean-based SQL injection attacks:
+
+```sh
+curl -X GET "http://httpbin.org/ip_route?id=1%27%20OR%201=1%20--" \
+     --header "accept: application/json"
+```
```
Comment (cloudjumpercat) [inline app/_hub/kong-inc/injection-protection/how-to/_index.md:64]: I'm having a hard time getting a validation step to work. ChatGPT helped me come up with some of this, but this is giving a 404. I also tried with localhost in the URL and that was a 404 as well. Any ideas what's going wrong here?

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+You can also configure custom regex matching.
+
+The Injection Protection plugin does the following: 
+* Extracts information from request headers, path/query parameters, or the payload body and evaluates that content against predefined regular expressions
+* Rejects the requests that match the regular expressions with a configurable HTTP status code and error message
+* Logs information about rejected requests for analytics and reporting
+
+## How does the Injection Protection plugin work?
+
+Depending on what you have configured in the plugin's config, the Injection Protection plugin does the following:
```
Comment (lena-larionova) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:18]: This line and line 11 say almost the same thing, but I think the difference is "what this plugin can do for you" (bad phrasing, but you get it) and "how this plugin literally works, in order of execution". Is that correct? If so, I'd rephrase to indicate that.

File: app/_hub/kong-inc/injection-protection/overview/_index.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+Logs are automatically collected when you enable the Injection Protection plugin. You can view the logs with the following options:
+
+* [{{site.base_gateway}} error log](/gateway/latest/production/logging/)
+* Log serializer. You can view these logs with the following plugins:
+    * [File Log](/hub/kong-inc/file-log/)
+    * [HTTP Log](/hub/kong-inc/http-log/)
+    * [Kafka Log](/hub/kong-inc/kafka-log/)
+    * [TCP Log](/hub/kong-inc/tcp-log/)
+    * [UDP Log](/hub/kong-inc/udp-log/)
+* [{{site.konnect_short_name}} Advanced Analytics](/konnect/analytics/) <!--though i'm not sure if they will do anything with them yet-->
```
Comment (lena-larionova) [inline app/_hub/kong-inc/injection-protection/overview/_index.md:82]:
```suggestion
* [{{site.konnect_short_name}} Advanced Analytics](/konnect/analytics/) <!--though i'm not sure if they will do anything with them yet--> 


## Get started with the Injection Protection plugin
* [Configuration reference](/hub/kong-inc/injection-protection/configuration/)
* [Basic configuration example](/hub/kong-inc/injection-protection/how-to/basic-example/)
```

**Review by lena-larionova — APPROVED** [review]

**Review by brentos — APPROVED** [review]
